src/os/bluestore/BlueStore.cc

```diff
@@ -4565,7 +4565,9 @@ int BlueStore::_open_db(bool create)
     string bfn;
     struct stat st;
 
-    bfn = path + "/block.db";
+    if (read_meta("path_block.db", &bfn) < 0) {
+      bfn = path + "/block.db";
+    }
```

Comment:
can we `write_meta("path_block.wal", bfn)` in this branch, so we can simplify it in the next release.

Reply:
we can't write_meta from _open_block() because in the mkfs case we haven't initialized the main label yet.  and it wouldn't help us for N because we need to go from L->N, and even though i expect to backport this they might upgrade from 12.2.0 or .1.  and ceph-disk is following a slightly different setup procedure...

I'm looking to further simplify this but I think it'll be a follow-up